### PR TITLE
Rework metadata handling to use the new test config structure for sta…

### PR DESCRIPTION
**Pull request by geoffjentry**
…ndard tests

Notes to reviewers
- Only a few tests are actually checking metadata (and thus outputs). Technically this is a regression but @ruchim is working to figure out the right values to drop in on all of these, so didn't seem worth the time to put it in now
- The Centaur code is complicated enough now that it should have self-tests, I plan on attacking that soon in #38 
- There's a fairly janky comparison between values supplied in the test file and from metadata which limits comparison to strings, numbers and booleans. At the moment that satisfies our needs, we can readdress when it does not
- I left a FIXME in there (you'll know it when you see it), I didn't want that to block getting this back over to Ruchi
- I apologize for all of the text file removals and moving code around
- I switched from using Lenthall's config wrapper to a library called `configs`, both because it made my life easier in a couple of spots and also as part of my larger desire to start doing away w/ lenthall

### CODE REVIEW

**Comment by geoffjentry** [comment]
@ruchim @kshakir

File: src/main/scala/centaur/test/Test.scala [hunk 1]
```diff
@@ -116,164 +113,42 @@ object Operations {
...
-
-    val diff = expectedMap.keySet -- actualMetadata.keySet
-
-    if (diff.nonEmpty) { println(s"For workflow ${request.name}: \nMissing expected $testType fields: $diff") }
-
-    expectedMap foreach { case (k, v: JsValue) =>
-      if (actualMetadata.contains(k) && (!v.toString.equals(actualMetadata.get(k).mkString))) {
-          println(s"Unexpected $testType for key: $k. Expected: ${v.toString} Actual: ${actualMetadata.get(k).mkString}")
-      }
+      override def run: Try[SubmittedWorkflow] = workflowLengthFutureCompletion(Future { doPerform() } )
```
Comment (kshakir) [inline src/main/scala/centaur/test/Test.scala]: Unbalanced spaces in `(Future { doPerform() } )`

File: src/main/scala/centaur/test/formulas/TestFormulas.scala [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    for {
+      _ <- runWorkflowUntilTerminalStatus(workflow, Succeeded)
+      _ <- runSuccessfulWorkflowAndVerifyMetadata(secondWorkflow)
+    } yield ()
+  }
+
+  def runCachingTurnedOffWorkflow(workflow: Workflow) = {
+    for {
+      testWf <- runWorkflowUntilTerminalStatus(workflow, Succeeded)
+      m <- retrieveMetadata(testWf)
```
Comment (kshakir) [inline src/main/scala/centaur/test/formulas/TestFormulas.scala]: Longer name than "m"?

File: README.md [hunk 3]
```diff
@@ -17,15 +17,19 @@ You can add your own tests to the test suite by adding `-Dcentaur.optionalTestPa
...
 files {
   wdl: path/to/wdl
   inputs: optional/path/to/inputs
   options: optional/path/to/options
-  metadata: optional/path/to/metadata
+}
+
+metadata {
+  fully.qualified.key.name1: VALUE1
+  fully.qualified.key.name2: VALU2
```
Comment (kshakir) [inline README.md]: `VALU2` -> `VALUE2`

**Comment by kshakir** [comment]
LGTM 👍 

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/44/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

File: src/main/scala/centaur/test/formulas/TestFormulas.scala (hunk 2 above)
Comment (geoffjentry) [inline src/main/scala/centaur/test/formulas/TestFormulas.scala]: Just because you've pleasantly put up with some of my nittier whitespace complaints!

**Comment by ruchim** [comment]
👍 Running it now, and I'll take note of things we can iterate upon without holding back this PR. Changes are easy to follow :)

[Approved with PullApprove](https://pullapprove.com/broadinstitute/centaur/pull-request/44/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=broadinstitute/centaur)

**Pull request merged**